src/components/forms/SignUpForm/SignUpForm.tsx

```diff
@@ -1,49 +1,50 @@
 import './SignUpForm.sass';
-import {useForm} from 'react-hook-form';
+import {FieldValues, Path, useForm} from 'react-hook-form';
 import BasicTextField from '@/components/inputs/BasicTextField/BasicTextField';
 import BasicBtn from '@/components/btns/BasicBtn/BasicBtn';
-import {signUpSchema} from '@/constants/signUp.validation';
+import {signUpSchema, ISignUp} from '@/constants/signUp.validation';
 import {yupResolver} from '@hookform/resolvers/yup';
+import {SignUpFields} from '@/models/signUp.enum';
 
-interface FormData {
-  firstName: string;
-  lastName: string;
-  email: string;
-  password: string;
-  phoneNumber?: string;
-}
+const fields = [
+  {name: 'firstName', label: 'First name'},
+  {name: 'lastName', label: 'Last name'},
+  {name: 'email', label: 'Email'},
+  {name: 'password', label: 'Password'},
+  {name: 'confirmPassword', label: 'Confirm password'},
+  {name: 'phoneNumber', label: 'Phone number'}
+];
 
 const SignUpForm: React.FC = () => {
   const {
     register,
     handleSubmit,
     control,
     formState: {errors}
-  } = useForm<FormData>({
+  } = useForm({
     resolver: yupResolver(signUpSchema)
   });
 
-  const onSubmit = (data: FormData): void => {
+  const onSubmit = (data: ISignUp): void => {
     console.log(data);
   };
 
+  const isPasswordField = (field: {name: Path<FieldValues>}): boolean => {
+    return field.name === 'password' || field.name === 'confirmPassword';
```

Comment:
It is better to have additional property `fieldType` and `enum` for it if you want to use such dynamic approach. Name isn`t stable.

Reply:
Thanks. 
```
const fields = [
  {name: 'firstName', label: 'First name', fieldType: FieldType.Text},
  {name: 'lastName', label: 'Last name', fieldType: FieldType.Text},
  {name: 'email', label: 'Email', fieldType: FieldType.Text},
  {name: 'password', label: 'Password', fieldType: FieldType.Password},
  {name: 'confirmPassword', label: 'Confirm password', fieldType: FieldType.Password},
  {name: 'phoneNumber', label: 'Phone number', fieldType: FieldType.Text}
];
```